### PR TITLE
Expose Grafana setting `viewers_can_edit` to helm chart (no ticket)

### PR DESCRIPTION
**Pull request by erdii**
* add grafana.provisioning.configuration.viewers_can_edit to grafana's values.yaml
* change grafana.ini to expose flag setting to grafana

**What this PR does / why we need it**: Let `Viewers` explore loki logs without removing the source-of-truth for grafana dashboard configuration.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: config/monitoring/grafana/config/grafana.ini [hunk 1]
```diff
@@ -14,7 +14,8 @@ header_name = X-Auth-Username
 header_property = username
 auto_sign_up = true
 
-{{ with .Values.grafana.provisioning.configuration.auto_assign_org_role }}
 [users]
+viewers_can_edit = {{ .Values.grafana.provisioning.configuration.viewers_can_edit }}
+{{- with .Values.grafana.provisioning.configuration.auto_assign_org_role }}
 auto_assign_org_role = {{ . }}
-{{ end }}
+{{- end }}
```
Comment (xrstf) [inline config/monitoring/grafana/config/grafana.ini]: Would you be so kind to improve this to just read

```
auto_assign_org_role = {{ .Values.grafana.provisioning.configuration.auto_assign_org_role }}
```

and remove the with...end statement? The value is always defined and now that we always have a `[user]` section, no need to be fancy.

**Review by xrstf — COMMENTED** [review]
Suggestion

**Comment by erdii** [comment]
@xrstf incorporated your feedback :muscle:

**Comment by xrstf** [comment]
Lovely

/lgtm
/approve

**Pull request merged**